examples/tutorials/device_avsr_tutorial.py

```diff
@@ -0,0 +1,526 @@
+    def __init__(self, segment_length: int, context_length: int, rate_ratio: int):
+        self.segment_length = segment_length
+        self.context_length = context_length
+
+        self.context_length_v = context_length
+        self.context_length_a = context_length * rate_ratio
+        self.context_v = torch.zeros([self.context_length_v, 3, 340, 600])
+        self.context_a = torch.zeros([self.context_length_a, 1])
+
+    def __call__(self, chunk_v, chunk_a):
+        if chunk_v.size(0) < self.segment_length:
+            chunk_v = torch.nn.functional.pad(chunk_v, (0, 0, 0, 0, 0, 0, 0, self.segment_length - chunk_v.size(0)))
+        if chunk_a.size(0) < self.segment_length * 640:
+            chunk_a = torch.nn.functional.pad(chunk_a, (0, 0, 0, self.segment_length * 640 - chunk_a.size(0)))
+
+        if self.context_length == 0:
+            return chunk_v.float(), chunk_a.float()
+        else:
+            chunk_with_context_v = torch.cat((self.context_v, chunk_v))
+            chunk_with_context_a = torch.cat((self.context_a, chunk_a))
+            self.context_v = chunk_v[-self.context_length_v :]
+            self.context_a = chunk_a[-self.context_length_a :]
+            return chunk_with_context_v.float(), chunk_with_context_a.float()
+
+
+######################################################################
+# 2. Pre-processing
+# -----------------
+#
+# Before feeding the raw stream into our model, each video sequence has to
+# undergo a specific pre-processing procedure. This involves three
+# critical steps. The first step is to perform face detection. Following
+# that, each individual frame is aligned to a referenced frame, commonly
+# known as the mean face, in order to normalize rotation and size
+# differences across frames. The final step in the pre-processing module
+# is to crop the face region from the aligned face image.
+#
+# .. raw:: html
+#
+#    <table align="center">
+#
+# .. raw:: html
+#
+#    <tr>
+#
+# .. raw:: html
+#
+#    <td>
+#
+# <img
+# src=“https://download.pytorch.org/torchaudio/doc-assets/avsr/original.gif”,
+# width=“144”>
+#
+# .. raw:: html
+#
+#    </td>
+#
+# .. raw:: html
+#
+#    <td>
+#
+# .. raw:: html
+#
+#    </td>
+#
+# .. raw:: html
+#
+#    <td>
+#
+# .. raw:: html
+#
+#    </td>
+#
+# .. raw:: html
+#
+#    <td>
+#
+# .. raw:: html
+#
+#    </td>
+#
+# .. raw:: html
+#
+#    <tr>
+#
+# .. raw:: html
+#
+#    <td>
+#
+# 0. Original
+#
+#    .. raw:: html
+#
+#       </td>
+#
+#    .. raw:: html
+#
+#       <td>
+#
+#    1. Detection
+#
+#       .. raw:: html
+#
+#          </td>
+#
+#       .. raw:: html
+#
+#          <td>
+#
+#       2. Transformation
+#
+#          .. raw:: html
+#
+#             </td>
+#
+#          .. raw:: html
+#
+#             <td>
+#
+#          3. Face ROIs
+#
+#             .. raw:: html
+#
+#                </td>
+#
+#             .. raw:: html
+#
+#                </tr>
+#
+#             .. raw:: html
+#
+#                </table>
+#
```

Comment:
It seems some gifs are missing.

Can you replace this with list-table directive?

https://sublime-and-sphinx-guide.readthedocs.io/en/latest/tables.html